src/lib/Scenes/Artwork/Components/CommercialButtons/BidButton.tsx

```diff
@@ -0,0 +1,154 @@
+  }
+
+  render() {
+    const { artwork } = this.props
+
+    if (artwork.sale && artwork.sale.isClosed) {
+      return null
+    }
+
+    const registrationAttempted = !!artwork.sale.registrationStatus
+    const registeredToBid = registrationAttempted && artwork.sale.registrationStatus.qualifiedForBidding
+
+    /**
+     * NOTE: This is making an incorrect assumption that there could only ever
+     *       be 1 live sale with this work. When we run into that case, there is
+     *       likely design work to be done too, so we can adjust this then.
+     */
+    const myLotStanding = artwork.myLotStanding && artwork.myLotStanding[0]
+    const hasMyBids = !!(myLotStanding && myLotStanding.mostRecentBid)
+
+    if (artwork.sale.isPreview) {
+      return (
+        <>
+          {!registrationAttempted && (
+            <Button width={100} block size="large" mt={1} onPress={() => this.redirectToRegister()}>
+              Register to bid
+            </Button>
+          )}
+          {registrationAttempted &&
+            !registeredToBid && (
+              <Button width={100} block size="large" mt={1} disabled>
+                Registration pending
+              </Button>
+            )}
+          {registrationAttempted &&
+            registeredToBid && (
+              <Button width={100} block size="large" mt={1} disabled>
+                Registration complete
+              </Button>
+            )}
+        </>
+      )
+    } else if (artwork.sale.isLiveOpen) {
+      return (
+        <>
+          {artwork.sale.isRegistrationClosed &&
+            !registeredToBid && (
+              <Sans size="2" color={color("black60")} pb={1} textAlign="center">
+                Registration closed
+              </Sans>
+            )}
+          <Button width={100} block size="large" onPress={() => this.redirectToLiveBidding()}>
+            {artwork.sale.isRegistrationClosed && !registeredToBid ? "Watch live bidding" : "Enter live bidding"}
+          </Button>
+        </>
+      )
+    } else if (registrationAttempted && !registeredToBid) {
+      return (
+        <Button width={100} block size="large" disabled>
+          Registration pending
+        </Button>
+      )
+    } else if (artwork.sale.isRegistrationClosed && !registeredToBid) {
+      return (
+        <Button width={100} block size="large" disabled>
+          Registration closed
+        </Button>
+      )
+    } else {
+      const myLastMaxBid = hasMyBids && myLotStanding.mostRecentBid.maxBid.cents
+      const increments = artwork.saleArtwork.increments.filter(increment => increment.cents > (myLastMaxBid || 0))
+      const firstIncrement = increments && increments.length && increments[0]
+      const incrementCents = firstIncrement && firstIncrement.cents
+
+      return (
+        <Button width={100} size="large" block onPress={() => this.redirectToBid(incrementCents)}>
+          {hasMyBids ? "Increase max bid" : "Bid"}
+        </Button>
+      )
```

Comment:
woah the logic in this method is hard to follow 😅 I don't know if it's possible to simplify so this is not a blocking comment. I'd be a bit scared to edit this code though haha.

Reply:
This was copied over from Reaction so gonna leave it as it reflects working logic.